include/clap/ext/draft/state-context.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "../../plugin.h"
+#include "../../stream.h"
+
+/// @page state-context extension
+/// @brief extended state handling
+///
+/// This extension let the host save and load the plugin state for special purpose.
+///
+/// Briefly, when loading a preset or duplicating a device, the plugin may want to partially load
+/// the state and initialize certain things differently.
+///
+/// Save and Load operations may have a different context.
+///
+/// Here is a conceptual implementation:
+/// state clap_plugin_state_context.save(ctx) {
+///    state = clap_plugin_state.save();
+///    return reduce(state, ctx);
+/// }
+///
+/// bool clap_plugin_state_context.load(state, ctx) {
+///    state = reduce(state, ctx);
+///    return clap_plugin_state.load(state);
+/// }
```

Comment:
Is this example necessary? I'm afraid a lot of people will not understand what this means without some additional context (heh). Just saying that the host may freely mix saving and loading operations from the state-context and state extensions and that the plugin should anticipate on that should be sufficient.